c9/logger_config.py

```
@@ -0,0 +1,200 @@
+def prepare_log_file(log_file: str | None, project_logger: logging.Logger) -> Path | None:
+    """
+    Prepare the log file path and create parent directories if needed.
+    """
+    if not log_file:
+        return None
+    log_file_path = os.path.expandvars(str(log_file))
+    log_file_path = Path(log_file_path).expanduser()
+
+    try:
+        log_file_path.parent.mkdir(parents=True, exist_ok=True)
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)